src/libraries/System.Text.Json/src/System/Text/Json/AppContextSwitchHelper.cs

```diff
@@ -5,9 +5,13 @@ namespace System.Text.Json
 {
     internal static class AppContextSwitchHelper
     {
-        public static bool IsSourceGenReflectionFallbackEnabled => s_isSourceGenReflectionFallbackEnabled;
+        public static bool IsDefaultReflectionDisabled { get; } =
```

Comment:
@vitek-karas I read in the feature switch documentation that static constructors should ideally be avoided. Does this need to be refactored in any way?

Reply:
I "think" this should be OK. I honestly don't understand the note about not using static .ctors. All of the properties we already have will need a piece of static .ctor anyway (autogenerated, but that doesn't really change anything).
@eerhardt maybe remembers some additional reasons.